### PR TITLE
Refactor the user model and add a user api endpoint

### PR DESCRIPTION
**Pull request by sampaccoud**
## Purpose

We want to start connecting users from the frontend to the backend.

## Proposal

- Refactor the user model to fit what is expected from the frontend
- Add an API endpoint for the user
- Add a full test suite for the new user API endpoint

### CODE REVIEW

File: src/magnify/apps/core/models.py [hunk 1]
```diff
@@ -22,15 +24,66 @@ def save(self, *args, **kwargs):
...
+        _("staff status"),
+        default=False,
+        help_text=_("Designates whether the user can log into this admin site."),
+    )
+    is_active = models.BooleanField(
+        _("active"),
+        default=True,
+        help_text=_(
+            "Designates whether this user should be treated as active. "
+            "Unselect this instead of deleting accounts."
```
[the remark this answers is not in the document]
Comment (sampaccoud) [reply src/magnify/apps/core/models.py:65]: This is Django default way of doing things. The thing is that sometimes it is complicated to delete a user with many related objects. In our case I think you're right, we want to allow deleting users.

File: src/magnify/apps/core/api.py [hunk 2]
```diff
@@ -31,6 +39,86 @@ def exception_handler(exc, context):
     return drf_exception_handler(exc, context)
 
 
+class UserViewSet(
+    mixins.CreateModelMixin,
+    mixins.DestroyModelMixin,
+    mixins.ListModelMixin,
+    mixins.RetrieveModelMixin,
+    mixins.UpdateModelMixin,
```
Comment (lunika) [inline src/magnify/apps/core/api.py:47]:
```suggestion
     viewsets.ModelViewSet
```

**Review by lunika — APPROVED** [review]

File: src/magnify/apps/core/api.py (hunk 2 above)
Comment (sampaccoud) [reply src/magnify/apps/core/api.py]: Can't be done anymore :wink:

**Comment by sampaccoud** [comment]
@lunika on the list endpoint I required to pass a search query and throttled the failed search attempts aggressively so one can not look for a username or email by repeated calls to this API.

**Pull request merged**